plugins/workflow_objc/src/activities/remove_memory_management.rs

```diff
@@ -0,0 +1,196 @@
+
+    // Remove any j_ prefix that the shared cache workflow adds to stub functions.
+    if symbol_name.starts_with_str("j_") {
+        symbol_name = &symbol_name[2..];
+    }
+
+    IGNORABLE_MEMORY_MANAGEMENT_FUNCTIONS.contains(&symbol_name)
+}
+
+fn process_instruction(
+    bv: &BinaryView,
+    llil: &LowLevelILFunction<Mutable, NonSSA>,
+    insn: &LowLevelILInstruction<Mutable, NonSSA>,
+    link_register: LowLevelILRegisterKind<CoreRegister>,
+    link_register_size: usize,
+) -> Result<bool, &'static str> {
+    if !is_call_to_ignorable_memory_management_function(bv, insn) {
+        return Ok(false);
+    }
+
+    // TODO: Removing calls to `objc_release` can sometimes leave behind a load of a struct field
+    // that appears to be unused. It's not clear whether we should be trying to detect and remove
+    // those here, or if some later analysis pass should be cleaning them up but isn't.
+
+    match insn.kind() {
+        LowLevelILInstructionKind::TailCall(_) => unsafe {
+            llil.set_current_address(insn.address());
+            llil.replace_expression(
+                insn.expr_idx(),
+                llil.ret(llil.reg(link_register_size, link_register)),
+            );
+        },
+        LowLevelILInstructionKind::Call(_) => unsafe {
+            // The memory management functions that are currently supported either return void
+            // or return their first argument. For arm64, the first argument is passed in `x0`
+            // and results are returned in `x0`, so we can replace the call with a nop. We'll need
+            // to revisit this to support other architectures, and to support the `objc_retain_x`
+            // `objc_release_x` functions that accept their argument in a different register.
+            llil.set_current_address(insn.address());
+            llil.replace_expression(insn.expr_idx(), llil.nop());
+        },
+        LowLevelILInstructionKind::Goto(_) if insn.index.0 == 0 => unsafe {
+            // If the `objc_retain` is the first instruction in the function, this function
+            // can only contain the call to the memory management function since when the
+            // memory management function returns, it will return to this function's caller.
+            llil.set_current_address(insn.address());
+            llil.replace_expression(
+                insn.expr_idx(),
+                llil.ret(llil.reg(link_register_size, link_register)),
+            );
+        },
+        LowLevelILInstructionKind::Goto(_) => {
+            // The shared cache workflow inlines calls to stub functions, which causes them
+            // to show up as a `lr = <next instruction>; goto <stub function instruction>;`
+            // sequence. We need to remove the load of `lr`  and update the `goto` to jump
+            // to the next instruction.
+
+            let Some(prev) =
+                llil.instruction_from_index(LowLevelInstructionIndex(insn.index.0 - 1_usize))
```

Comment:
```suggestion
                llil.instruction_from_index(LowLevelInstructionIndex(insn.index.0 - 1))
```

isn't it more readable that way?

Reply:
Good point, this is a big readability improvement.